app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt

```diff
@@ -1,54 +1,114 @@
+            selectedDate = LocalDate.of(year, month + 1, day)
+        }
+    }
+
+    /*
+    This function allows the user to add a new deadline directly, using the "ENTER" key (more intuitive).
+     */
+    private fun initTextInput() {
+        val edit = findViewById<TextInputEditText>(R.id.textInputEditCalendar)
+        edit.setOnKeyListener { v, keycode, event ->
+            if (keycode == KeyEvent.KEYCODE_ENTER && event.action == KeyEvent.ACTION_DOWN) {
+                // Enter is a new shortcut to add a deadline (intuitive)
+                addNewDeadlineCalendar(v)
+
+                // Hide the keyboard and clear the focus on the text input
+                edit.clearFocus()
+                val view = this.currentFocus
+                if (view != null) {
+                    val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+                    imm.hideSoftInputFromWindow(view.windowToken, 0)
+                }
+                // The listener has consumed the event
+                return@setOnKeyListener true
+            }
+            false
+        }
+    }
+
+    /*
+    This function allows the user to exit the text input intuitively, just by clicking outside
+     */
+    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
+        if (event.action == MotionEvent.ACTION_DOWN) {
+            // We are in the case were the user has touched outside
+            val v = currentFocus
+            if (v is EditText) {
+                val outRect = Rect()
+                v.getGlobalVisibleRect(outRect)
+                if (!outRect.contains(event.rawX.toInt(), event.rawY.toInt())) {
+                    // If the user has touched a place outside the keyboard, remove the focus and keyboard
+                    v.clearFocus()
+                    val imm = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
+                    imm.hideSoftInputFromWindow(v.getWindowToken(), 0)
+                }
+            }
+        }
+        return super.dispatchTouchEvent(event)
     }
 
     /*
     The purpose of this function is to provide the user the ability to add a new
     deadline easily using directly the calendar to select the date.
-    ***TO DO***: store deadline and display them on the calendar.
      */
     fun addNewDeadlineCalendar(view: View) {
         // Getting the necessary views
         val editText = findViewById<TextInputEditText>(R.id.textInputEditCalendar)
-        val calendar = findViewById<CalendarView>(R.id.calendar_view)
 
         // Getting the entered text and the selected date
         val deadlineTitle = editText.text.toString()
-        val selectedDate =
-            Instant.ofEpochMilli(calendar.date).atZone(ZoneId.systemDefault()).toLocalDate()
 
         // Add the new event to the deadline list
         val deadline = Deadline(deadlineTitle, DeadlineState.TODO, selectedDate)
+
         viewModel.addDeadline(deadline)
```

Comment:
I see that you use now the viewModel instead of the repository. Why this change ?